src/app/categories/state/entry-categories.service.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the &quot;License&quot;);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an &quot;AS IS&quot; BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Injectable } from '@angular/core';
+import { Observable, Subscription } from 'rxjs';
+import { Category, EntriesService } from '../../shared/openapi';
+import { EntryCategoriesStore } from './entry-categories.store';
+import { EntryCategoriesQuery } from './entry-categories.query';
+
+@Injectable({ providedIn: 'root' })
+export class EntryCategoriesService {
+  public categories$: Observable<Array<Category>>;
+
+  constructor(
+    private entryCategoriesStore: EntryCategoriesStore,
+    private entryCategoriesQuery: EntryCategoriesQuery,
+    private entriesService: EntriesService,
+  ) {
+    /**
+     * An observable list of the categories for a given entry.
+     */
+    this.categories$ = this.entryCategoriesQuery.selectAll();
+  }
+
+  /**
+   * Current entry category update subscription.
+   */
+  private currentSubscription: Subscription = null;
+
+  /**
+   * Updates the list of categories for the specified entry.
+   */
+  updateEntryCategories(entryId: number) {
+    this.entryCategoriesStore.setLoading(true);
+    this.entryCategoriesStore.setError(false);
+    this.entryCategoriesStore.remove();
+    this.currentSubscription?.unsubscribe();
```

Comment:
If `updateEntryCategories` were invoked with two different entry ids in succession, the response for the first entry could get unsubscribed prematurely, and its response ignored. But it looks like the store only has categories for one entry anyway, it may not matter and might actually be the desired behavior.

Reply:
What you describe is the desired behavior.  I implemented this service/store similarly to the existing services/stores, which store a single unit of information which ostensibly corresponds to an item on the current page.  Essentially, my little tweak ignores the first response if it arrives after another request is launched, because we've changed/updated the page and don't want/need the first response any more.

I have some ideas about dynamically-allocated stores which don't suffer from this race condition, and would allow us to do things like retrieve the categories for each of multiple entries on the same page, but that's a discussion for some other day...